src/pages/Trace/Instructions/Instructions.tsx

```diff
@@ -1,19 +1,49 @@
 import React, { useRef, useEffect } from 'react';
+import { AppModels } from '../../../canvas/App';
 import { NavButton } from 'components/NavButton/NavButton';
-import { InstructionModal } from 'components/InstructionModal/InstructionModal';
+import { Modal } from 'components/Modal/Modal';
 import { PathTypeButtonGrid } from 'components/PathTypeButtonGrid/PathTypeButtonGrid';
+import { PatternPathType, ModalType } from 'canvas/Enums';
+import { ActionButton } from 'components/ActionButton/ActionButton';
 import './Instructions.css';
-import { PatternPathType } from 'canvas/Enums';
+
 
 interface InstructionsProps {
     curPathType: PatternPathType;
     setPathType: React.Dispatch<React.SetStateAction<PatternPathType>>;
 }
 
 export const Instructions: React.FC<InstructionsProps> = ({curPathType, setPathType}) => {
-    const doneButton: NavButton = {label: 'DONE', to: 'AddMeasurement' };
+    const showNavButton: NavButton = {label: 'DONE', to: 'AddMeasurement' };
     const addPathButton: NavButton = {label:'', to: 'AddPath'};
-    const instruction: InstructModal = {text: ['Choose seam, fold, or edge to add to your pattern']};
+    const warning: Modal = {text: ['Your pattern is incomplete. Please add another line.'], type: ModalType.Warning};
+    const instruction: Modal = {text: ['Choose seam, fold, or edge to add to your pattern'], type: ModalType.Instruction};
+    
+    const [showWarning, setShowWarning] = React.useState<boolean>(false);
+    let doneContainer = <></>;
+    const showDoneButton = useRef(AppModels.document.isEmpty());
+    const arePatternsEnclosed = useRef(AppModels.document.arePatternPiecesEnclosed());
+
+    useEffect(() => {
+        showDoneButton.current = AppModels.document.isEmpty();
+        arePatternsEnclosed.current = AppModels.document.arePatternPiecesEnclosed();
+    }, [arePatternsEnclosed, showDoneButton]);
+
+    const handleClickDone = (): void => {
+        setShowWarning(!AppModels.document.arePatternPiecesEnclosed());
+    };
+    const showWarningButton: ActionButton = {label: 'DONE', action: handleClickDone};
```

Comment:
I'm a bit confused at why this button is named showWarning.

Reply:
There are two buttons on the instruction page that look the same, one is showWarning and the other is showNavButton. Depending on the conditions, the nav button is shown, or the warning button is shown. If it makes more sense to you, I will change them to 'warningButton and navButton'.

Reply:
yeah I'd like the "show..." to be booleans and the buttons named "...Button". I feel that would help a lot with readability. But more than that, I don't get what showWarningButton has to do with warnings since it displays "done" and handles the click normally.

Reply:
AAAAAAhh I think I get it: it's a "decide if we show warning" button. is that it?

Reply:
Yeah, that's what it is. We already know that you can't be done, so clicking it just displays the warning message. But I'll rename them to what I mentioned earlier. It's a bit easier to understand.

Reply:
I made some major changes to the buttons. Please see the Refactor Buttons commit for all of the changes. Basically I removed The NavButton and ActionButton from our types and updated the NavButton and ActionButton to use Button.

Reply:
I like the refactor of the buttons. for the Instructions code I left a new comment below.